pkg/devfile/adapters/kubernetes/utils/utils_test.go

```diff
@@ -168,62 +284,86 @@ func TestUpdateContainersWithSupervisord(t *testing.T) {
 								Name: component,
 							},
 						},
+						{
+							Container: &versionsCommon.Container{
+								Name: debugComponent,
+							},
+						},
 					},
 					ExecCommands: tt.execCommands,
 				},
 			}
 
-			containers, err := UpdateContainersWithSupervisord(devObj, tt.containers, tt.runCommand)
+			containers, err := UpdateContainersWithSupervisord(devObj, tt.containers, tt.runCommand, tt.debugCommand, 5858)
 
 			if !tt.wantErr && err != nil {
 				t.Errorf("TestUpdateContainersWithSupervisord unxpected error: %v", err)
 			} else if tt.wantErr && err != nil {
 				// return since we dont want to test anything further
 				return
+			} else if tt.wantErr && err == nil {
+				t.Error("wanted error but got not error")
 			}
 
 			// Check if the supervisord volume has been mounted
 			supervisordVolumeMountMatched := false
 			envRunMatched := false
 			envWorkDirMatched := false
+			envDebugMatched := false
+			envDebugWorkDirMatched := false
 
 			if tt.execCommands[0].WorkingDir == "" {
 				// if workdir is not present, dont test for matching the env
 				envWorkDirMatched = true
 			}
 
+			if len(tt.execCommands) >= 2 && tt.execCommands[1].WorkingDir == "" {
+				// if workdir is not present, dont test for matching the env
+				envDebugWorkDirMatched = true
+			}
+
 			for _, container := range containers {
-				if container.Name == component {
-					for _, volumeMount := range container.VolumeMounts {
-						if volumeMount.Name == adaptersCommon.SupervisordVolumeName && volumeMount.MountPath == adaptersCommon.SupervisordMountPath {
-							supervisordVolumeMountMatched = true
+				for _, testContainer := range tt.containers {
+					if container.Name == testContainer.Name {
+						for _, volumeMount := range container.VolumeMounts {
+							if volumeMount.Name == adaptersCommon.SupervisordVolumeName && volumeMount.MountPath == adaptersCommon.SupervisordMountPath {
+								supervisordVolumeMountMatched = true
+							}
 						}
-					}
 
-					for _, envVar := range container.Env {
-						if envVar.Name == adaptersCommon.EnvOdoCommandRun && envVar.Value == tt.execCommands[0].CommandLine {
-							envRunMatched = true
-						}
-						if tt.execCommands[0].WorkingDir != "" && envVar.Name == adaptersCommon.EnvOdoCommandRunWorkingDir && envVar.Value == tt.execCommands[0].WorkingDir {
-							envWorkDirMatched = true
+						for _, envVar := range container.Env {
+							if envVar.Name == adaptersCommon.EnvOdoCommandRun && envVar.Value == tt.execCommands[0].CommandLine {
+								envRunMatched = true
+							}
+							if tt.execCommands[0].WorkingDir != "" && envVar.Name == adaptersCommon.EnvOdoCommandRunWorkingDir && envVar.Value == tt.execCommands[0].WorkingDir {
+								envWorkDirMatched = true
+							}
+							if len(tt.execCommands) >= 2 && envVar.Name == adaptersCommon.EnvOdoCommandDebug && envVar.Value == tt.execCommands[1].CommandLine {
```

Comment:
Please add comments to explicit what this is testing and group similar condition checks under the same `if` where possible.